src/benchmarking/benchmark.py

```diff
@@ -0,0 +1,617 @@
+        else:
+            print('Format of experiment condition title is not supported. Example: "10_genomes_1000000_reads".')      
+
+        return num_genomes, num_reads
+    
+
+    def parse_simulated_fasta(self, ncbi_to_taxid_dict: Dict) -> Dict[str, float]:
+        '''
+        DESCRIPTION:
+            Parses 'truth' sequencing file and stores tax IDs and corresponding abundances
+            in a dictionary
+        INPUT:
+            -
+        OUTPUT:
+            Assigns and returns taxid<>abundance key-value pair in self.abundances_dict
+        '''
+
+        total_counts = 0
+        not_found = set()
+        prev_len = 0
+        try:
+            fasta_opened = open(self.sequencing_file, "r")
+        except OSError as e:
+            print(f'Problem opening {self.sequencing_file}')
+            return self.abundances_dict # return empty dictionary
+        else:
+            line = fasta_opened.readline()
+            while (line):
+                if (line[0] == ">"):
+                    ncbi_id = line[1:].split("|")[0].split("-")[0]
+                    # convert NCBI to taxonomic ID
+                    if ncbi_id in ncbi_to_taxid_dict:
+                        tax_id = ncbi_to_taxid_dict[ncbi_id]
+                    else: 
+                        #print(f"NCBI ID {ncbi_id} not in ncbi2taxid dictionary")
+                        file_name = line[1:].split("|")[-1].strip("\n")
+                        try:
+                            path = Path(settings.PHAGE_REF_DB_PATH + file_name)
+                        except AttributeError as e:
+                            print(f"Unable to open 'PHAGE_REF_DB_PATH': {e}", file=sys.stderr)
+                        else:
+                            try:
+                                tax_id = open(path).readline().split("kraken:taxid|")[1].strip("\n").replace(" ", "")
+                            except:
+                                not_found.add(path)
+                                new_len = len(not_found)
+                                if new_len != prev_len:
+                                    print(f"ERROR (parsing):")
+                                    print(f"\tTool: simulated files")
+                                    print(f"\t\tnumber of true NOT found (fine if 1): {new_len}")
+                                    print(f"\t\tNCBI ID: {ncbi_id}")
+                                    prev_len = new_len
+                    if tax_id in self.abundances_dict:
+                        self.abundances_dict[tax_id] += 1
+                    else:
+                        self.abundances_dict[tax_id] = 1
+                    total_counts += 1
+                line = fasta_opened.readline()
+            fasta_opened.close()
+            
+        #turn counts into abundances - normalize
+        for tax_id in self.abundances_dict.keys():
+            self.abundances_dict[tax_id] /= total_counts 
+
+        return self.abundances_dict
+
+
+class Result(ABC):
+
```

Comment:
Using the strategy pattern here is great. I really think this is cool! 

I can't think of a better method for parsing - perhaps mixing with a factory pattern, which is kind of what's done when choosing which parser.